contracts/andromeda_gumball/src/contract.rs

```diff
@@ -0,0 +1,881 @@
+pub const PERIOD: u64 = 30;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    CW721_CONTRACT.save(deps.storage, &msg.andromeda_cw721_contract)?;
+    // Set initial status to false since there's nothing to buy upon instantiation
+    let new_list: Vec<String> = Vec::new();
+    LIST.save(deps.storage, &new_list)?;
+    STATUS.save(deps.storage, &false)?;
+    ADOContract::default().instantiate(
+        deps.storage,
+        deps.api,
+        info,
+        BaseInstantiateMsg {
+            ado_type: "gumball".to_string(),
+            operators: None,
+            modules: None,
+            primitive_contract: None,
+        },
+    )
+}
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    match msg {
+        ExecuteMsg::AndrReceive(msg) => contract.execute(deps, env, info, msg, execute),
+        ExecuteMsg::Mint(mint_msg) => execute_mint(deps, env, info, mint_msg),
+        ExecuteMsg::Buy {} => execute_buy(deps, env, info),
+        ExecuteMsg::SaleDetails {
+            price,
+            max_amount_per_wallet,
+            recipient,
+        } => execute_sale_details(deps, env, info, price, max_amount_per_wallet, recipient),
+        ExecuteMsg::SwitchStatus {} => execute_switch_status(deps, info),
+        // ExecuteMsg::SetContractAddress {
+        //     andromeda_cw721_contract,
+        // } => execute_switch_contract_address(deps, info, andromeda_cw721_contract),
+    }
+}
+// fn execute_switch_contract_address(
+//     deps: DepsMut,
+//     info: MessageInfo,
+//     msg: AndrAddress,
+// ) -> Result<Response, ContractError> {
+//     let contract = ADOContract::default();
+
+//     require(
+//         contract.is_contract_owner(deps.storage, info.sender.as_str())?,
+//         ContractError::Unauthorized {},
+//     )?;
+//     CW721_CONTRACT.save(deps.storage, &msg)?;
+//     Ok(Response::new().add_attribute("action", "set cw721 address"))
+// }
+fn execute_switch_status(deps: DepsMut, info: MessageInfo) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    let mut status = STATUS.load(deps.storage)?;
+    require(
+        contract.is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+    // in case owner forgot to set the state, can't allow purchasing without the sale details set
+    let state = STATE.may_load(deps.storage)?;
+    require(state.is_some(), ContractError::PriceNotSet {})?;
+    // Automatically switch to opposite status
+    if status {
+        status = false;
+    } else {
+        status = true;
+    }
```

Comment:
nit: Can be simplified with `status = !status`

Reply:
Very elegant!